crates/p2p/src/signer.rs

```diff
@@ -0,0 +1,22 @@
+//! Application signer trait for P2P message signing.
+//!
+//! This module provides the ApplicationSigner trait that allows service
+//! libraries to provide signing functionality for messages without requiring
+//! strata-p2p to store private keys.
+
+use std::fmt::Debug;
+
+use libp2p::identity::PublicKey;
+
+/// Trait for signing setup messages with application private keys.
+///
+/// The implementation should use the keypair that corresponds to the provided app_public_key.
```

Comment:
```suggestion
/// The implementation should use the keypair that corresponds to the provided `app_public_key`.
```

small docstring nit.